gst/datarepo/gstdatareposrc.c

```diff
@@ -526,66 +543,218 @@ gst_data_repo_src_read_tensors (GstDataRepoSrc * src, GstBuffer ** buffer)
       if user sets "tensor-sequence=2,1", datareposrc read offset 9528 then 9488.
     */
 
-    data = info[i].data;
+    data = info.data;
 
     byte_read = 0;
     to_read = src->tensors_size[seq_idx];
     offset = sample_offset + src->tensors_offset[seq_idx];
-    src->offset = lseek (src->fd, offset, SEEK_SET);
+    src->fd_offset = lseek (src->fd, offset, SEEK_SET);
 
     while (to_read > 0) {
       GST_LOG_OBJECT (src,
           "Reading %d bytes at offset 0x%" G_GINT64_MODIFIER "x (%d size)",
-          to_read, src->offset + byte_read, (guint) src->offset + byte_read);
+          to_read, src->fd_offset + byte_read,
+          (guint) src->fd_offset + byte_read);
       errno = 0;
-      ret = read (src->fd, data + byte_read, to_read);
-      GST_LOG_OBJECT (src, "Read: %d", ret);
-      if (ret < 0) {
+      read_size = read (src->fd, data + byte_read, to_read);
+      GST_LOG_OBJECT (src, "Read: %d", read_size);
+      if (read_size < 0) {
         if (errno == EAGAIN || errno == EINTR)
           continue;
-        goto could_not_read;
+        GST_ELEMENT_ERROR (src, RESOURCE, READ, (NULL), GST_ERROR_SYSTEM);
+        ret = GST_FLOW_ERROR;
+        goto error;
       }
       /* files should eos if they read 0 and more was requested */
-      if (ret == 0) {
+      if (read_size == 0) {
         /* .. but first we should return any remaining data */
         if (byte_read > 0)
           break;
-        goto eos;
+        GST_DEBUG ("EOS");
+        ret = GST_FLOW_EOS;
+        goto error;
       }
-      to_read -= ret;
-      byte_read += ret;
+      to_read -= read_size;
+      byte_read += read_size;
 
-      src->read_position += ret;
-      src->offset += ret;
+      src->read_position += read_size;
+      src->fd_offset += read_size;
     }
 
-    if (mem[i])
-      gst_memory_unmap (mem[i], &info[i]);
+    if (mem)
+      gst_memory_unmap (mem, &info);
 
-    gst_buffer_append_memory (buf, mem[i]);
+    gst_buffer_append_memory (buf, mem);
   }
 
   *buffer = buf;
 
   return GST_FLOW_OK;
 
-could_not_read:
-  {
-    GST_ELEMENT_ERROR (src, RESOURCE, READ, (NULL), GST_ERROR_SYSTEM);
-    gst_memory_unmap (mem[0], &info[0]);
-    gst_buffer_unref (buf);
-    return GST_FLOW_ERROR;
+error:
+  gst_memory_unmap (mem, &info);
```

Comment:
suggestion: leak case if failed to map memory at line 518 mem = gst_allocator_alloc();

```
GstMemory *mem = NULL;
...

error:
  if (mem) {
    gst_memory_unmap (mem, &info);
    gst_memory_unref (mem);
  }

  gst_buffer_unref (buf);
  return ret;
```

Reply:
```mem``` does not need to ```gst_memory_unref (mem)```.

Reply:
Okay, I got it!!, thank you.